src/compiler/checker.ts

```diff
@@ -10663,7 +10730,17 @@ namespace ts {
                 });
             }
 
-            forEach(node.members, checkSourceElement);
+            forEach(node.members, node.flags & NodeFlags.Abstract ?
+                checkSourceElement :
+                element => {
+                    checkSourceElement(element);
+
+                    // Classes containing abstract methods must be marked abstract
+                    if (element.flags & NodeFlags.Abstract) {
+                        error(node, Diagnostics.Classes_containing_abstract_methods_must_be_marked_abstract);

```

Comment:
Yes, this constitutes reporting the same error.

The grammar check can be modified to report the error on the class declaration, but then we might report the same error on the class multiple times (ie a non-abstract has multiple abstract methods).